nilearn/decomposition/_base.py

```diff
@@ -478,12 +480,19 @@ def fit(self, imgs, y=None, confounds=None):
             )
 
         masker_type = "nii"
-        if isinstance(self.mask, (SurfaceMasker, SurfaceImage)) or any(
-            isinstance(x, SurfaceImage) for x in imgs
+        if (
+            isinstance(self.mask, (SurfaceMasker, SurfaceImage))
+            or any(isinstance(x, SurfaceImage) for x in imgs)
+            or any(
+                isinstance(x, str) and x.endswith(SURFACE_DATA_EXTENSIONS)
+                for x in imgs
+            )
```

Comment:
@Remi-Gau do we have a better way to check the file type via file extensions or paths.

I hit the same issue in concat_imgs.

Reply:
Ah ok it's because we never directly provide the paths to .gii images. 
I forgot that loading surface image needs to be done by putting together a mesh and data files (accessed via separate gii files):

https://nilearn.github.io/stable/auto_examples/00_tutorials/plot_surface_101.html#load-the-surface-image

Reply:
yup for the moment we never instanciate SurfaceImage directly from disk